tools/osbuild-image-info

```diff
@@ -2567,6 +2571,52 @@ def partition_is_lvm(part: Dict) -> bool:
     return part["type"].upper() in ["E6D6D379-F507-44C2-A23C-238F2A3DF928", "8E"]
 
 
+def find_fstab_root(tree, fstype):
+    """
+    Find the root volume under tree by searching for /etc/fstab.
+    This function first checks if the path <tree>/etc/fstab exists and then, if the fstype is btrfs, checks all
+    subvolumes as well.
+
+    Returns None if fstab is not found.
+    """
+    if os.path.exists(f"{tree}/etc/fstab"):
+        return tree
+
+    if fstype == "btrfs":
+        root_subvol = find_root_subvol(tree)
+        if root_subvol:
+            return root_subvol
+
+    return None
+
+
+def find_root_subvol(root):
+    """
+    Given a btrfs volume root, find the subvolume that contains the root OS tree.
+    """
+    def subvol_list_parser(output):
+        """
+        Parse the output of 'btrfs subvolume list' and return just the subvolume names/paths.
+        """
+        paths = []
+        for line in output.strip().split("\n"):
+            # subvolume names can have spaces in them, but they are the last field and they are preceded by the word
+            # path
+            parts = line.split(" ")  # last component is the path
+            path_word_index = parts.index("path")
+            path = " ".join(parts[path_word_index+1:])
+            paths.append(path)
+        return paths
+    subvols = subprocess_check_output(["btrfs", "subvolume", "list", root], parse_fn=subvol_list_parser)
+
+    # look through each subvol for /etc/fstab
+    for subvol in subvols:
+        path = os.path.join(root, subvol)
+        if os.path.exists(os.path.join(path, "etc/fstab")):
+            return path
+
```

Comment:
Maybe an explicit `return None` here at the end of the function (iirc pylint also warns about those missing).

Reply:
Missed that, thanks.  Strange that I didn't see a warning from pylint about it locally.  I think the embedded function/closure might be throwing it off, which lead me to realise it doesn't need to be a closure.

Anyway, fixing now.